ports/rp2/machine_i2c.c

```diff
@@ -97,7 +97,7 @@ static void machine_i2c_print(const mp_print_t *print, mp_obj_t self_in, mp_prin
 mp_obj_t machine_i2c_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
     enum { ARG_id, ARG_freq, ARG_scl, ARG_sda, ARG_timeout };
     static const mp_arg_t allowed_args[] = {
-        { MP_QSTR_id, MP_ARG_REQUIRED | MP_ARG_OBJ },
+        { MP_QSTR_id,  MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
```

Comment:
Still not ideal but I'd probably do:

```python
#ifdef PICO_DEFAULT_I2C
    { MP_QSTR_id,  MP_ARG_INT, {.u_int = PICO_DEFAULT_I2C} },
#else
    { MP_QSTR_id,  MP_ARG_INT, {.u_int = -1} },
#endif
```

Rather than inline the checks below. Saves code.

(Not really sure why this was ever `MP_ARG_OBJ`?)

It doesn't feel like this breaks the API, but it seems to break expectations on Pico since - afaict - `machine.I2C()` will not give you a properly configured default I2C instance. (It doesn't handle `PICO_DEFAULT_I2C_SCL_PIN` and `PICO_DEFAULT_I2C_SDA_PIN` so I2C will wiggle into the void and thus your objective - portable code - wont quite be met?)